ISSUER_PROTOCOL.md

```diff
@@ -0,0 +1,334 @@
+
+
+## Trust Token Crypto Protocol
+
+The Trust Token crypto protocol is based on the PMBTokens design in https://eprint.iacr.org/2020/072 (appendix H) using P-384. The necessary keys and function mappings are described below. Serialization follows the TLS presentation language (https://tools.ietf.org/html/rfc8446).
+
+
+### Keys
+
+The Trust Token protocol primarily requires two kinds of keypairs:
+
+*   SRRSigningKey/SRRVerificationKey - A Ed25519 public keypair for signing and verifying the integrity of the Signed Redemption Record response.
+*   TrustTokenSecretKey/TrustTokenPublicKey - A Trust Token keypair used to sign and verify Trust Tokens.
+
+#### SRRSigningKey/SRRVerificationKey
+
+These keys should be generated and stored as standard Ed25519 keys, with the public/verification key being included in the Key Commitment result as a 32-byte RFC8032 encoding.
+
+
+#### TrustTokenSecretKey/TrustTokenPublicKey
+
+These are keys used in Trust Token consisting of elliptic curve scalars and points. All scalars and points are sized based on the curve choice (P-384). Up to 3 keys may be configured in parallel in the key commitment.
+
+Each keypair consist of the following:
+
+
+```
+{
+  Scalar x0, y0; // Corresponding to the FALSE private metadata bit (x0, y0).
+  Scalar x1, y1; // Corresponding to the TRUE private metadata bit (x1, y1).
+  Scalar xs, ys; // Corresponding to the token validity (x˜, y˜).
+} TrustTokenSecretKey;
+{
+  ECPoint pub0; // Corresponding to the FALSE private metadata bit.
+  ECPoint pub1; // Corresponding to the TRUE private metadata bit.
+  ECPoint pubs; // Corresponding to the token validity check.
+} TrustTokenPublicKey;
+```
+
+The TrustTokenPublicKey is encoded in the key commitment and should be serialized as follows:
+
+
+```
+{
+  opaque pub0<1..2^16-1>; // X9.62 Uncompressed point.
+  opaque pub1<1..2^16-1>; // X9.62 Uncompressed point.
+  opaque pubs<1..2^16-1>; // X9.62 Uncompressed point.
+} TrustTokenPublicKey;
+```
+
+### Serialization/Hashing
+
+For the PMBTokens functions, the following serialization schemes and hashes are used internally using draft 07 of the hash-to-curve specification (https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-07):
+
+
+```
+Ht(t) - P384_XMD:SHA-512_SSWU_RO_ with a big-endian bytestring input of 't' and a dst of "PMBTokens Experiment V1 HashT".
+
+Hs(T, s) - p384_xmd_sha512_sswu with the following input and a dst of "PMBTokens Experiment V1 HashS".
+
+{
+  opaque t<1..2^16-1>; // X9.62 Uncompressed point.
+  opaque s[Nn];
+};
+
+Hc(x) - p384_xmd_sha512 scalar derivation with the input x and a dst of "PMBTokens Experiment V1 HashC". Depending on the context, one of the following serializations is used for the input:
```

Comment:
Define p384_xmd_sha512 scalar derivation, since we made it up?

Reply:
Done.